core/src/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.cpp

```diff
@@ -47,21 +47,31 @@ namespace {
         logger::error() << msg << std::endl; \
     }
 
+#define LOG_WARN(msg)                       \
+    if (_config.verbose_error) {            \
+        logger::warn() << msg << std::endl; \
+    }
+
 // This logger is specific to the greedy_ambiguity_resolution_algorithm, and to
 // this translation unit.
 struct logger {
     static std::ostream& error() {
-        std::cout << "ERROR @greedy_ambiguity_resolution_algorithm: ";
+        std::cout << "ERROR: @greedy_ambiguity_resolution_algorithm: ";
+        return std::cout;
+    }
+
+    static std::ostream& warn() {
+        std::cout << "WARNING: @greedy_ambiguity_resolution_algorithm: ";
         return std::cout;
     }
 
     static std::ostream& info() {
-        std::cout << "@greedy_ambiguity_resolution_algorithm: ";
+        std::cout << "INFO: @greedy_ambiguity_resolution_algorithm: ";
         return std::cout;
     }
 
     static std::ostream& flood() {
-        std::cout << "@greedy_ambiguity_resolution_algorithm: ";
+        std::cout << "INFO: @greedy_ambiguity_resolution_algorithm: ";
         return std::cout;
     }
```

Comment:
I'm generally not the biggest fan of all of these output messages. :thinking: But as long as we use them, why not name this as either a `DEBUG` or `VERBOSE` output level? To be a bit more in line with Athena / Python / etc. message logging conventions.

Reply:
Thanks for your review :)

I made a mistake, I meant to say `FLOOD:` for "flood" messages. I completely agree with you, I think it's much better to have a common vocabulary and not confuse developers with too many different names referring to the same concept. I’ve changed `FLOOD` into `DEBUG` (since I only used these messages for debugging purposes). They are here to help me in case anything goes wrong and I need to debug the code, but I agree, the fewer messages we print on the terminal, the better, and there is no point in repeatedly saying "I'm doing this and that, everything works fine", better save space for the really important messages.